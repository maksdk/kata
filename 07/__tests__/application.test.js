//@ts-check
import fs from 'fs';
import path from 'path';
import {
    html
} from 'js-beautify';
import timer from 'timer-promise';

import run from '../src';

const htmlOptions = {
    preserve_newlines: true,
    unformatted: [],
};

const fixturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => html(document.body.innerHTML, htmlOptions);

beforeEach(() => {
    const initHtml = fs.readFileSync(path.join(fixturesPath, 'index.html')).toString();
    document.documentElement.innerHTML = initHtml;
    run();
});

test('application', async () => {
    expect(getTree()).toMatchSnapshot();

    const [nameLink, valueLink] = document.querySelectorAll('a');

    nameLink.click();
    await timer.start(10);
    expect(getTree()).toMatchSnapshot();

    valueLink.click();
    await timer.start(10);
    expect(getTree()).toMatchSnapshot();

    valueLink.click();
    await timer.start(10);
    expect(getTree()).toMatchSnapshot();

    nameLink.click();
    await timer.start(10);
    expect(getTree()).toMatchSnapshot();

    valueLink.click();
    await timer.start(10);
    expect(getTree()).toMatchSnapshot();
});

test('application 2', async () => {
    expect(getTree()).toMatchSnapshot();
});