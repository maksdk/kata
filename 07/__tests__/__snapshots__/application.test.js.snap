// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">
    <table class=\\"table\\">
        <tbody>
            <tr>
                <th><a href=\\"#\\">Name (Asc)</a></th>
                <th><a href=\\"#\\">Value (Unsorted)</a></th>
            </tr>
            <tr>
                <td>host</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>hostname</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>href</td>
                <td>http://localhost/</td>
            </tr>
            <tr>
                <td>origin</td>
                <td>http://localhost</td>
            </tr>
            <tr>
                <td>pathname</td>
                <td>/</td>
            </tr>
            <tr>
                <td>protocol</td>
                <td>http:</td>
            </tr>
        </tbody>
    </table>
</div>"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">
    <table class=\\"table\\">
        <tbody>
            <tr>
                <th><a href=\\"#\\">Name (Desc)</a></th>
                <th><a href=\\"#\\">Value (Unsorted)</a></th>
            </tr>
            <tr>
                <td>protocol</td>
                <td>http:</td>
            </tr>
            <tr>
                <td>pathname</td>
                <td>/</td>
            </tr>
            <tr>
                <td>origin</td>
                <td>http://localhost</td>
            </tr>
            <tr>
                <td>href</td>
                <td>http://localhost/</td>
            </tr>
            <tr>
                <td>hostname</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>host</td>
                <td>localhost</td>
            </tr>
        </tbody>
    </table>
</div>"
`;

exports[`application 2 1`] = `
"<div class=\\"container m-3\\">
    <table class=\\"table\\">
        <tbody>
            <tr>
                <th><a href=\\"#\\">Name (Asc)</a></th>
                <th><a href=\\"#\\">Value (Unsorted)</a></th>
            </tr>
            <tr>
                <td>host</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>hostname</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>href</td>
                <td>http://localhost/</td>
            </tr>
            <tr>
                <td>origin</td>
                <td>http://localhost</td>
            </tr>
            <tr>
                <td>pathname</td>
                <td>/</td>
            </tr>
            <tr>
                <td>protocol</td>
                <td>http:</td>
            </tr>
        </tbody>
    </table>
</div>"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">
    <table class=\\"table\\">
        <tbody>
            <tr>
                <th><a href=\\"#\\">Name (Unsorted)</a></th>
                <th><a href=\\"#\\">Value (Asc)</a></th>
            </tr>
            <tr>
                <td>pathname</td>
                <td>/</td>
            </tr>
            <tr>
                <td>protocol</td>
                <td>http:</td>
            </tr>
            <tr>
                <td>origin</td>
                <td>http://localhost</td>
            </tr>
            <tr>
                <td>href</td>
                <td>http://localhost/</td>
            </tr>
            <tr>
                <td>host</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>hostname</td>
                <td>localhost</td>
            </tr>
        </tbody>
    </table>
</div>"
`;

exports[`application 4`] = `
"<div class=\\"container m-3\\">
    <table class=\\"table\\">
        <tbody>
            <tr>
                <th><a href=\\"#\\">Name (Unsorted)</a></th>
                <th><a href=\\"#\\">Value (Desc)</a></th>
            </tr>
            <tr>
                <td>host</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>hostname</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>href</td>
                <td>http://localhost/</td>
            </tr>
            <tr>
                <td>origin</td>
                <td>http://localhost</td>
            </tr>
            <tr>
                <td>protocol</td>
                <td>http:</td>
            </tr>
            <tr>
                <td>pathname</td>
                <td>/</td>
            </tr>
        </tbody>
    </table>
</div>"
`;

exports[`application 5`] = `
"<div class=\\"container m-3\\">
    <table class=\\"table\\">
        <tbody>
            <tr>
                <th><a href=\\"#\\">Name (Asc)</a></th>
                <th><a href=\\"#\\">Value (Unsorted)</a></th>
            </tr>
            <tr>
                <td>host</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>hostname</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>href</td>
                <td>http://localhost/</td>
            </tr>
            <tr>
                <td>origin</td>
                <td>http://localhost</td>
            </tr>
            <tr>
                <td>pathname</td>
                <td>/</td>
            </tr>
            <tr>
                <td>protocol</td>
                <td>http:</td>
            </tr>
        </tbody>
    </table>
</div>"
`;

exports[`application 6`] = `
"<div class=\\"container m-3\\">
    <table class=\\"table\\">
        <tbody>
            <tr>
                <th><a href=\\"#\\">Name (Unsorted)</a></th>
                <th><a href=\\"#\\">Value (Asc)</a></th>
            </tr>
            <tr>
                <td>pathname</td>
                <td>/</td>
            </tr>
            <tr>
                <td>protocol</td>
                <td>http:</td>
            </tr>
            <tr>
                <td>origin</td>
                <td>http://localhost</td>
            </tr>
            <tr>
                <td>href</td>
                <td>http://localhost/</td>
            </tr>
            <tr>
                <td>host</td>
                <td>localhost</td>
            </tr>
            <tr>
                <td>hostname</td>
                <td>localhost</td>
            </tr>
        </tbody>
    </table>
</div>"
`;
